specs/WebMessageObjects.md

```diff
@@ -171,18 +402,82 @@ namespace Microsoft.Web.WebView2.Core
         /// Any DOM object type that can be natively representable that has been passed in to
         /// `additionalObjects` parameter will be accessible here.
         /// Currently a WebMessage object can be the following type:
-        /// $net$
-        /// - System.IO.FileInfo
-        /// $end$
-        /// $winrt$
-        /// - Windows.Storage.StorageFile
-        /// $end$
+        /// - CoreWebView2File
+        /// - CoreWebView2FileSystemHandle
         /// Cast the object to the native type to access its specific properties.
         [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2WebMessageReceivedEventArgs2")]
         {
             IVectorView<Object> AdditionalObjects { get; };
         }
     }
+
+    runtimeclass CoreWebView2
+    {
+        ...
+        /// Same as PostWebMessageAsJson, but also has support for posting DOM objects
+        /// to page content. The `additionalObjects` property in the DOM MessageEvent
+        /// fired on the page content is a ArrayLike list of
+        /// DOM objects that can be posted to the web content. Currently the
+        /// following types can be posted to DOM:
+        /// | .NET/WinRT       | DOM type    |
+        /// |------------------|-------------|
+        /// | CoreWebView2File | [File](https://developer.mozilla.org/docs/Web/API/File] |
+        /// | CoreWebView2FileSystemHandle | [FileSystemHandle](https://developer.mozilla.org/docs/Web/API/FileSystemHandle) |
+        /// | null             | null        |
+        /// The objects are posted the to web content following the structured-clone
+        /// semantics, meaning only objects that can be cloned can be posted.
+        /// They will also behave as in they are created on the web content they are
+        /// posted to. For example, if a FileSystemFileHandle is posted to a web content
+        /// it can only be posted via postMessage to other web content with the same origin.
+        PostWebMessageAsJsonWithAdditionalObjects(string webMessageAsJson, IVectorView<Object> additionalObjects);

```

Comment:
Should this have an `[overload("PostWebMessageAsJson")]` so that it is projected with a cleaner name?

Reply:
Or maybe it's named this way to align with the JS API postMessageWithAdditionalObjects, which is itself not an overload because there is a pre-existing overload on the JS side that takes a transferList.